src/rmp/test/helpers.tcl

```diff
@@ -0,0 +1,89 @@
+# Helper functions common to multiple regressions.
+if {[package vcompare [package present Tcl] 8.6] == -1} {
+    proc lmap {args} {
+        set result {}
+        set var [lindex $args 0]
+        foreach item [lindex $args 1] {
+            uplevel 1 "set $var $item"
+            lappend result [uplevel 1 [lindex $args end]]
+        }
+        return $result
+    }
+}
+
+set test_dir [file dirname [file normalize [info script]]]
+set result_dir [file join $test_dir "results"]
```

Comment:
look at all of the regression tests that use the function make_result_file defined in helpers.tcl, which you should link, not copy